ai/mistralai/v0/main.go

```diff
@@ -0,0 +1,119 @@
+//go:generate compogen readme ./config ./README.mdx
+package mistralai
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"sync"
+
+	"github.com/instill-ai/component/base"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+const (
+	TextGenerationTask = "TASK_TEXT_GENERATION_CHAT"
+	TextEmbeddingTask  = "TASK_TEXT_EMBEDDINGS"
+	cfgAPIKey          = "api-key"
+)
+
+var (
+	//go:embed config/definition.json
+	definitionJSON []byte
+	//go:embed config/setup.json
+	setupJSON []byte
+	//go:embed config/tasks.json
+	tasksJSON []byte
+
+	once sync.Once
+	comp *component
+)
+
+type component struct {
+	base.Component
+	instillAPIKey string
+}
+
+func Init(bc base.Component) *component {
+	once.Do(func() {
+		comp = &component{Component: bc}
+		err := comp.LoadDefinition(definitionJSON, setupJSON, tasksJSON, nil)
+		if err != nil {
+			panic(err)
+		}
+	})
+	return comp
+}
+
+type execution struct {
+	base.ComponentExecution
+	execute                func(*structpb.Struct) (*structpb.Struct, error)
+	client                 MistralClient
+	usesInstillCredentials bool
+}
+
+// WithInstillCredentials loads Instill credentials into the component, which
+// can be used to configure it with globally defined parameters instead of with
+// user-defined credential values.
+func (c *component) WithInstillCredentials(s map[string]any) *component {
+	c.instillAPIKey = base.ReadFromGlobalConfig(cfgAPIKey, s)
+	return c
+}
+
+func (c *component) CreateExecution(sysVars map[string]any, setup *structpb.Struct, task string) (*base.ExecutionWrapper, error) {
+	resolvedSetup, resolved, err := c.resolveSetup(setup)
+	if err != nil {
+		return nil, err
+	}
+	e := &execution{
+		ComponentExecution:     base.ComponentExecution{Component: c, SystemVariables: sysVars, Task: task, Setup: resolvedSetup},
+		client:                 newClient(getAPIKey(setup), c.GetLogger()),
```

Comment:
Question
About InstillCredentials part, do you just copy and paste?
If so, could you help confirm this `newClient(getAPIKey(setup)`?
I saw other parts using `newClient(getAPIKey(resolvedSetup)`
I am not sure which one is correct.

And, same question to Cohere component.

Reply:
Fixed, it should be newClient(getAPIKey(resolvedSetup). I think we can have a unified function to get the api keys and resolve instill-credit-keys for different AI components. We can follow convention over configuration methodology and assume all api-keys are stored in the  `api-key` field ? Let me know your thought.

Reply:
@namwoam 

> I think we can have a unified function to get the api keys and resolve instill-credit-keys for different AI components

Yes! I think so. 
Because this is not a blocker for this PR, let's change it in your next PR!